VeganVirus/VeganVirus/SoundAction.cpp

```diff
@@ -2,42 +2,46 @@
 
 SoundAction::SoundAction(double req) : Action(req, MUSIC_ICON_PATH)
 {
-    playListTime[0] = 62;
+    _timer = 0;
 }
 
 void SoundAction::start()
 {
-    if (!canActivate())
+    _timer = SOUND_DUARTION;
+}
+
+void SoundAction::update(double dt)
+{
+    std::string song = "";
+    _timer -= dt;
+    if (_timer <= 0 || !canActivate())
     {
         return;
     }
-    std::string song = "";
+    
     //change the volume to 30 for now
-    this->_audioManager.changeVolume(30);
+    this->_audioManager.changeVolume(60);
 
     //choose randomly song
     this->currentSong = rand() % NUM_OF_SONGS;
     song = "play music\\song" + std::to_string(this->currentSong) + ".wav";
 
     //play and check when song start
     this->_audioManager.play(song);
-    this->songStartTime = clock();
```

Comment:
This should be inside `start` and not update. you only ever start a new song when start is called, and there are no more cases.